### PR TITLE
BUG: stats: fix discrete `.isf` to work at boundaries when `loc != 0`

### LINKED ISSUE #14236: BUG: discrete isf is wrong at boundary if loc != 0
I'm writing unit tests for statsmodels geometric and found a bug

#11248 fixed ppf for loc != 0
but AFAICS isf was never fixed

I think only the corner case is wrong, (my unit tests against geom loc = -1 pass in other cases)

```
import scipy
scipy.__version__
'1.6.1'

# expected behavior if support starts at zero
stats.poisson.isf(1, 1.), stats.poisson.isf(0, 1.)
(-1.0, inf)

# support in default starts at 1
stats.geom.isf(1, 1.), stats.geom.isf(0, 1.)
(0.0, inf)

# BUG did not shift to take loc into account
stats.geom.isf(1, 1., loc=-1), stats.geom.isf(0, 1., loc=-1)
(0.0, inf)
```

### PR DESCRIPTION
**Pull request by tirthasheshpatel**
#### Reference issue

closes gh-14236

#### What does this implement/fix?

The `.isf` method of discrete distributions has been changed to incorporate location shift for quantiles `0` and `1` as explained in gh-14236. Appropriate tests have also been added. This PR closely resembles gh-11248.

#### Additional information

N/A

### CODE REVIEW

File: scipy/stats/tests/test_discrete_basic.py [hunk 1]
```diff
@@ -148,6 +148,31 @@ def test_ppf_with_loc(dist, args):
...
+    except TypeError:
+        distfn = dist
+    # check with a negative, no and positive relocation.
+    np.random.seed(1942349)
+    re_locs = [np.random.randint(-10, -1), 0, np.random.randint(1, 10)]
+    _a, _b = distfn.support(*args)
+    for loc in re_locs:
+        expected = _b + loc, _a - 1 + loc
+        res = distfn.isf(0., *args, loc=loc), distfn.isf(1., *args, loc=loc)
+        npt.assert_allclose(expected, res, rtol=1e-14)
```
Comment (josef-pkt) [inline scipy/stats/tests/test_discrete_basic.py:164]: this should also work with asserting equality. expected and res should be integer valued

**Comment by josef-pkt** [comment]
Thanks, looks good to me

File: scipy/stats/tests/test_discrete_basic.py (hunk 1 above)
Comment (tirthasheshpatel) [reply scipy/stats/tests/test_discrete_basic.py]: Changed it to assert equality. Thanks!

**Review by rgommers — APPROVED** [review]
LGTM too. And nice test (I checked it's fast - can be a problem for stats distributions test). Thanks @tirthasheshpatel. And thanks for the review @josef-pkt
